pkg/testutils/fake.go

```diff
@@ -0,0 +1,256 @@
+/*
+Copyright 2021 The Cockroach Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testutils
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+// NewFakeClient returns a new fake client
+func NewFakeClient(scheme *runtime.Scheme, objs ...runtime.Object) *FakeClient {
+	return &FakeClient{
+		client: fake.NewFakeClientWithScheme(scheme, objs...),
+		scheme: scheme,
+	}
+}
+
+var _ client.Client = &FakeClient{}
+
+// FakeClient is inspired and taken from https://github.com/kubernetes/client-go/
+type FakeClient struct {
+	client client.Client
+	scheme *runtime.Scheme
+
+	// ReactionChain is the list of reactors that will be attempted for every
+	// request in the order they are tried.
+	ReactionChain []Reactor
+}
+
+// Reactor is an interface to allow the composition of reaction functions.
+type Reactor interface {
```

Comment:
Why are we recreating https://pkg.go.dev/k8s.io/client-go/testing#Reactor

Reply:
Again this `Reactor` part is from cockroach-operator. It's not used at the moment in this self-signer the testing. Kept it here for future references/usage. We can remove it once now or remove it when the utility is ready and this code part is unused.

Reply:
Makes sense